What's new in pytest 3.0

nicoddemus opened the pull request:
Still a work in progress, I plan to publish only after pytest-3.0 is effectively out.

Meanwhile, suggestions on writing style, order of the sessions, anything really, is welcome!

cc @hackebrot

nicoddemus committed:
WIP: First version of the "what's new in pytest 3.0" post

The-Compiler commented:
You seem to use double backticks in various places, which is ReST and not markdown :wink:

The-Compiler commented on content/posts/2016/whats-new-in-pytest-30.md:
```diff
@@ -0,0 +1,222 @@
+    def add_np(doctest_namespace):
+        doctest_namespace['np'] = numpy
+
+Doctests in the same below the `conftest.py` file can now use the `numpy` module directly:
```
"in the same below the `conftest.py` file"?

nicoddemus replied:
Fixed, thanks

The-Compiler commented on content/posts/2016/whats-new-in-pytest-30.md:
```diff
@@ -0,0 +1,222 @@
+
+Doctests in the same below the `conftest.py` file can now use the `numpy` module directly:
+
+    # content of numpy.py
```
Probably want a `:::python` here as well (is that the right syntax for whatever markdown dialect we're using here?)

nicoddemus replied:
It is according to the [documentation](http://docs.getpelican.com/en/3.6.3/content.html#syntax-highlighting), but TBH it did not work for me locally.

flub commented on content/posts/2016/whats-new-in-pytest-30.md:
```diff
@@ -0,0 +1,222 @@
+
+This blog post provides a short overview of some of the major features and changes in pytest 3.0. See the [CHANGELOG](http://doc.pytest.org/en/latest/changelog.html) for the complete list.
+
+# New entry point: ``pytest``
```
I find "entry point" a rather funny name as it reminds me too much of setuptools entrypoints.  Maybe something like "Additional command alias".  That also emphasises more that the "py.test" is not going anywhere (which I obviously like ;-)).

RonnyPfannschmidt replied:
i propose "new script name"

flub replied:
"additional script name"

yay, we're bikeshedding!  :)

nicoddemus replied:
I went with @flub's suggestion because it sounded better to me, sorry @RonnyPfannschmidt! :blush:

nicoddemus committed:
Add invocation-scoped fixtures and small fixes

nicoddemus commented:
> You seem to use double backticks in various places, which is ReST and not markdown

Yep, thanks! :confounded:

nicoddemus committed:
Add --doctest-report command line option

nicoddemus committed:
Remove invocation scoped fixtures section

As discussed in pytest-dev/pytest#1794, it was decided to
postpone this until 3.1

nicoddemus committed:
Remove section about docs restructuring

Unfortunately this won't land in 3.0

nicoddemus commented:
It seems like it should be ready to be posted.

Does anybody mind if I merge this myself after publishing the package?

The-Compiler commented:
Not sure how up-to-date it is, though:
- seems like the resultlog deprecation isn't in it (but that was just merged :laughing:)
- if we talk about deprecated features, should we also talk about removed commandline options?
- anything else?

nicoddemus committed:
Mention --resultlog was deprecated

hackebrot commented:
@nicoddemus 🍰

nicoddemus commented:
Hmm for some reason it did not appear yet on blog.pytest.org... :confused:

nicoddemus commented:
@pfctdayelise @pelme blog.pytest.org still doesn't show the new post. The [build](https://travis-ci.org/pytest-dev/blog.pytest.org/builds/153668656) appears to have been successful, any ideas?

nicoddemus commented:
Nevermind, the post was marked as a _draft_. Published and online now. :grin:
